test/config-luatest/memory_test.lua

```diff
@@ -0,0 +1,29 @@
+local yaml = require('yaml')
+local t = require('luatest')
+local treegen = require('test.treegen')
+local helpers = require('test.config-luatest.helpers')
+local justrun = require('test.justrun')
+
+local g = helpers.group()
+
+g.test_relative_config_path = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
```

Comment:
I suppose that running an additional instance for this test is overkill. We might just use `t.assert_error_msg_equals()` like you do for `test_lua_memory_setting()`.

Also, it may be done as a subtest in the <lua_memory_test.lua>, but I don't insist on it.

Reply:
When error occurs the Tarantool panics, as I found. `t.assert_error_msg_equals()` aren't able to handle it.

Probably it shouldn't work like this. I couldn't yet find a way to throw an error in allocator (either long jumps nor exceptions nor returning NULL from it works properly).

Reply:
> I couldn't yet find a way to throw an error in allocator

I suppose, in the case of a limit violation, the allocator function should just return `NULL`. In that case, the LuaJIT VM will raise the OOM error, as expected.

Reply:
It's implemented like this right now. I wasn't able to capture the error with `t.assert_error_msg_equals()`

Reply:
This is due to [not restored](https://github.com/tarantool/tarantool/pull/10223#discussion_r1704975669) `allocated` value in the case of the OOM.
After fixing the behaviour we are able to see last error via [`box.error.last()`](https://www.tarantool.io/ru/doc/2.11/reference/reference_lua/box_error/last/).

```sh
src/tarantool -e '
  local function bigalloc()
    -- On your branch before fixing the default value.
    ("x"):rep(1024 * 1024 * 256)
  end
  local r, e = pcall(bigalloc)
  print("err:",  box.error.last())
'
OutOfMemory: Failed to allocate 268435456 bytes in luaT_alloc for lua objects
err:    Failed to allocate 268435456 bytes in luaT_alloc for lua objects

```

Reply:
Oh, you're right. Fixed it. Thanks